src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingWriteStream.cs

```diff
@@ -30,13 +33,6 @@ public override Task WriteAsync(ReadOnlyMemory<byte> source, CancellationToken i
                     return _connection.FlushAsync();
                 }
 
-                if (_connection._currentRequest == null)
```

Comment:
There is a comment below in WriteChunkAsync about duplex communication.  At the very least, the comment should be removed.  Possibly the logic there should change too, though that may be beyond the scope of this PR.

Reply:
I removed the comment and the explicit flushing.  We do a flush at the end of copying the content; that should be sufficient, and of course any flushes that happen along the way as the buffer fills.  I've also changed ContentLengthWriteStream to use HttpConnection.WriteAsync instead of HttpConection.WriteWithoutBufferingAsync, and opened an issue for the CONSIDER comment in WriteWithoutBufferingAsync (WriteAsync will also skip the buffer if the writes are sufficiently large).